src/pages/ConvertVrt/index.tsx

```diff
@@ -53,15 +75,83 @@ export const ConvertVrtUi = ({
   );
 };
 
-const ConvertVrt = () => (
-  <ConvertVrtUi
-    xvsTotalWei={new BigNumber('300000000000000000')}
-    xvsToVrtRate={new BigNumber('00.0003')}
-    vrtLimitUsedWei={new BigNumber(300)}
-    vrtLimitWei={new BigNumber(1000)}
-    vrtConversionEndTime="1678859525"
-    walletConnected={false}
-  />
-);
+const ConvertVrt = () => {
+  const { account } = useContext(AuthContext);
+  const { t } = useTranslation();
+  const accountAddress = account?.address;
+  const { data: vrtConversionEndTime } = useGetVrtConversionEndTime();
+  const { data: vrtConversionRatio } = useGetVrtConversionRatio();
+  const { data: userVrtAllowance } = useGetAllowance(
+    {
+      tokenId: VRT_ID,
+      accountAddress: accountAddress || '',
+      spenderAddress: getContractAddress('vrtConverterProxy'),
+    },
+    { enabled: !!accountAddress },
+  );
+  const { data: userVrtBalanceWei } = useGetBalanceOf(
+    { accountAddress: accountAddress || '', tokenId: VRT_ID },
+    { enabled: !!accountAddress },
+  );
+  const { data: xvsVestedBalanceWei } = useGetBalanceOf(
+    { accountAddress: getContractAddress('xvsVestingProxy'), tokenId: 'xvs' },
+    { enabled: !!accountAddress },
+  );
+  const { data: { totalWithdrawableAmount: xvsWithdrawableAmount } = {} } =
+    useGetXvsWithdrawableAmount(
+      { accountAddress: accountAddress || '' },
+      { enabled: !!accountAddress },
+    );
+
+  const { mutateAsync: convertVrt, isLoading: convertVrtLoading } = useConvertVrt();
+  const { mutateAsync: withdrawXvs, isLoading: withdrawXvsLoading } = useWithdrawXvs();
+  const userVrtEnabled = new BigNumber(userVrtAllowance || 0).gt(0);
+
+  const handleConvertVrt = async (amount: string) => {
+    if (!accountAddress) {
+      throw new UiError(t('errors.walletNotConnected'));
+    }
+    const res = await convertVrt({
+      amountWei: amount,
+      accountAddress,
+    });
+    return res.transactionHash;
+  };
+
+  const handleWithdrawXvs = async () => {
+    if (!accountAddress) {
+      throw new UiError(t('errors.walletNotConnected'));
+    }
+    const res = await withdrawXvs({
+      accountAddress,
+    });
+    return res.transactionHash;
+  };
+
+  const conversionRatio = useMemo(() => {
+    if (vrtConversionRatio) {
+      return new BigNumber(vrtConversionRatio).div(CONVERSION_RATIO_DECIMAL);
+    }
+    return undefined;
+  }, [vrtConversionRatio]);
+  if (conversionRatio && userVrtBalanceWei && vrtConversionEndTime && xvsVestedBalanceWei) {
```

Comment:
In the case where no wallet is connected, the loading spinner will display indefinitely because `xvsVestedBalanceWei` and `userVrtBalanceWei` are `undefined`.

I think all the side effects should be handled by the individual tabs, which would solve that issue since they're the ones displaying `EnableToken` and `ConnectWallet`.